prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/http/RemoteWriteRetryStrategy.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.apache.hc.core5.util.TimeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLException;
+
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.net.ConnectException;
+import java.net.NoRouteToHostException;
+import java.net.UnknownHostException;
+import java.util.List;
+
+/** Retry strategy for the http client. */
+public class RemoteWriteRetryStrategy implements HttpRequestRetryStrategy {
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteWriteRetryStrategy.class);
+
+    private static final List<Class<? extends IOException>> NON_RETRIABLE_EXCEPTIONS =
+            List.of(
+                    InterruptedIOException.class,
+                    UnknownHostException.class,
+                    ConnectException.class,
+                    NoRouteToHostException.class,
+                    SSLException.class);
+
+    private final long initialRetryDelayMs;
+    private final long maxRetryDelayMs;
+    private final int maxRetryCount;
+
+    private final SinkMetrics counters;
+
+    public RemoteWriteRetryStrategy(RetryConfiguration retryConfiguration, SinkMetrics counters) {
+        this.initialRetryDelayMs = retryConfiguration.getInitialRetryDelayMS();
+        this.maxRetryDelayMs = retryConfiguration.getMaxRetryDelayMS();
+        this.maxRetryCount = retryConfiguration.getMaxRetryCount();
+        this.counters = counters;
+    }
+
+    @Override
+    public boolean retryRequest(
+            HttpRequest httpRequest, IOException e, int execCount, HttpContext httpContext) {
+        // Retry on any IOException except those considered non-retriable
+        var retry =
+                (execCount <= maxRetryCount) && !(NON_RETRIABLE_EXCEPTIONS.contains(e.getClass()));
+        LOG.debug(
+                "{} retry on {}, at execution {}",
+                (retry) ? "DO" : "DO NOT",
+                e.getClass(),
+                execCount);
+        countRetry(retry);
+        return retry;
+    }
+
+    @Override
+    public boolean retryRequest(HttpResponse httpResponse, int execCount, HttpContext httpContext) {
+        var retry =
+                (execCount <= maxRetryCount)
+                        && RemoteWriteResponseClassifier.isRetriableErrorResponse(httpResponse);
+        LOG.debug(
+                "{} retry on response {} {}, at execution {}",
+                (retry) ? "DO" : "DO NOT",
+                httpResponse.getCode(),
+                httpResponse.getReasonPhrase(),
+                execCount);
+        countRetry(retry);
+        return retry;
+    }
+
+    @Override
+    public TimeValue getRetryInterval(
+            HttpResponse httpResponse, int execCount, HttpContext httpContext) {
+        long calculatedDelay = initialRetryDelayMs << (execCount - 1);
```

Comment:
Should multiplier be configurable instead of being set to 2?

Reply:
I don't see much value in making the backoff strategy more sophisticated. 
Prometheus Remote-write specs just say it "MUST use a backoff algorithm to prevent overwhelming the server".
Tweaking the `initialRetryIntervalMs` and `maxRetryDelayMs` will be enough to make the strategy more or less aggressive.
Also, I don't see much value in adding a jitter. There is one writer thread per subtask, and and it's very unlikely they will be synchronized when writing to the endpoint.